internal/validator/storage.go

```diff
@@ -77,12 +78,13 @@ func (this *storageConnector) fetchPayload(url string) (payload []byte, err erro
 }
 
 func readFile(reader io.Reader) (result []byte, err error) {
-	var isGzip bool
 	reader = bufio.NewReaderSize(reader, 2)
+	compression, err := utils.GetCompressionType(reader)
+	if err != nil {
+		return nil, err
+	}
 
-	if isGzip, err = utils.IsGzip(reader); err != nil {
-		return
-	} else if isGzip {
+	if compression == "gzip" {
```

Comment:
Now that we have the enum defined it should be used here also. I am going to change that as I merge this.